demos/nrf52-writeable-attributes/src/main.rs

```diff
@@ -0,0 +1,370 @@
+//! Starts a GATT server that exposes a single attribute which, when written to,
+//! toggles an LED on/off (on pin 17 by default).
+#![no_std]
+#![no_main]
+
+#[cfg(feature = "52810")]
+use nrf52810_hal as hal;
+#[cfg(feature = "52832")]
+use nrf52832_hal as hal;
+#[cfg(feature = "52840")]
+use nrf52840_hal as hal;
+
+use core::cmp;
+use core::sync::atomic::{compiler_fence, Ordering};
+use hal::gpio::{Level, Output, Pin, PushPull};
+use hal::prelude::OutputPin;
+use rtt_target::{rprintln, rtt_init_print};
+use rubble::{
+    att::{AttUuid, Attribute, AttributeAccessPermissions, AttributeProvider, Handle, HandleRange},
+    config::Config,
+    l2cap::{BleChannelMap, L2CAPState},
+    link::{
+        ad_structure::AdStructure,
+        queue::{PacketQueue, SimpleQueue},
+        LinkLayer, Responder, MIN_PDU_BUF,
+    },
+    security::NoSecurity,
+    time::{Duration, Timer},
+    uuid::{Uuid128, Uuid16},
+    Error,
+};
+use rubble_nrf5x::{
+    radio::{BleRadio, PacketBuffer},
+    timer::BleTimer,
+    utils::get_device_address,
+};
+
+pub struct LedBlinkAttrs {
+    // Attributes exposed to clients that don't change.
+    // This includes the "primary service" and "characteristic" attributes.
+    static_attributes: [Attribute<&'static [u8]>; 3],
+    // State and resources to be modified/queried when packets are received.
+    // The AttributeValueProvider allows attributes to be generated lazily; those attributes should
+    // use these fields.
+    led_pin: Pin<Output<PushPull>>,
+    led_buf: [u8; 1],
+}
+
+const PRIMARY_SERVICE_UUID16: Uuid16 = Uuid16(0x2800);
+const CHARACTERISTIC_UUID16: Uuid16 = Uuid16(0x2803);
+const GENERIC_ATTRIBUTE_UUID16: Uuid16 = Uuid16(0x1801);
+
+// TODO what UUID should this be? I took this from a course assignment :P
```

Comment:
Any suggestions? I couldn't find any resources regarding how 128-bit UUIDs get determined.

Reply:
I think you're supposed to randomly generate one and stick to that